ethdb/table.go

```diff
@@ -118,12 +118,7 @@ func (t *Table) Close() error {
 		}
 	}
 	if t.segments != nil {
-		for _, segment := range t.segments.Slice() {
-			if err := segment.Close(); err != nil {
-				return err
-			}
-			t.segments.Remove(segment.Name())
-		}
+		t.segments.Close()
```

Comment:
There were a few problems with the old version:
- failed fast on first error and abandon any remaining segments
- didn't look for the Purge() method like onEvict, which frees mapped memory for file segments
- repeatedly grabbed the same locks and acquired the same semaphore for each segment removal